### ISSUE TITLE
Add `responders{type = "escalation"}` in `opsgenie_alert_policy`

### ISSUE THREAD

**Issue by janhorstmann** [opsgenie/terraform-provider-opsgenie#293]
Hi,
this is a feature request to support 
```hcl
  responders {
    type = "escalation"
}
```
in resource `opsgenie_alert_policy`.
Although the [API documentation states](https://docs.opsgenie.com/docs/alert-and-notification-policy-api#create-policy) that only `user` and `team` are possible values to the `responders` field, the web interface definitely allows setting an `escalation`. When doing so the `terraform-provider-opsgenie` correctly identifies the `type = "escalation"` as configuration drift:
```
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply":

  # opsgenie_alert_policy.test has changed
  ~ resource "opsgenie_alert_policy" "test" {
        id                         = SOME_ID
        name                       = "test alert policy"
        tags                       = []
        # (14 unchanged attributes hidden)


      ~ responders {
            id   = SOME_ID
          ~ type = "team" -> "escalation"
        }
        # (1 unchanged block hidden)
    }
```
Please add `responders{type = "escalation"}` in `opsgenie_alert_policy`.

### Affected Resource(s)
- opsgenie_alert_policy

### Terraform Configuration Files
```hcl
resource "opsgenie_alert_policy" "test" {
  name               = "example policy"
  team_id            = opsgenie_team.test.id
  policy_description = "This is sample policy"
  message            = "{{message}}"

  filter {}
  responders {
    type = "escalation"
    id = opsgenie_escalation.some_resource.id
  }
}
```

### Output
```
Error: expected responders.0.type to be one of [user team], got escalation
│ 
│   with opsgenie_alert_policy.test,
│   on main.tf line 336, in resource "opsgenie_alert_policy" "test":
│  336:     type = "escalation"
```

**Comment by ghost** [comment]
I'm not familiar with the code base, but I believe this is defined here:

https://github.com/opsgenie/terraform-provider-opsgenie/blob/0156e4721329a915b97c55203519c6af878340fd/opsgenie/resource_opsgenie_alert_policy.go#L514

```golang
		responder := alert.Responder{
			Type:     alert.ResponderType(config["type"].(string)),
			Id:       responderID,
			Name:     name,
			Username: username,
		}
```

**Comment by janhorstmann** [comment]
>  Although the [API documentation states](https://docs.opsgenie.com/docs/alert-and-notification-policy-api#create-policy) that only user and team are possible values to the responders field, the web interface definitely allows setting an escalation.

I have just tested the API and it seems to be possible to set a `type: escalation` in `responders`:  
```shell
#TEAM_ID=""
#ESCALATION_ID=""
#OPSGENIE_API_KEY=""

ALERT_POLICY_ID="$( curl -X POST \
                   https://api.eu.opsgenie.com/v2/policies?teamId=$TEAM_ID \
                   --header "Authorization: GenieKey $OPSGENIE_API_KEY" \
                   --header 'Content-Type: application/json' \
                   --data \
'{
    "continue": false,
    "type": "alert",
    "name": "test",
    "enabled": false,
    "policyDescription": "test",
    "filter": {
      "type": "match-all"
    },
    "responders": [
      {
        "type": "escalation",
        "id": "'$ESCALATION_ID'"
      }
    ],
    "alias": "{{alias}}",
    "message": "{{message}}",
    "description": "{{description}}",
    "source": "{{source}}",
    "entity": "{{entity}}",
    "actions": [],
    "tags": [],
    "details": {},
    "ignoreOriginalActions": false,
    "ignoreOriginalDetails": false,
    "ignoreOriginalResponders": true,
    "ignoreOriginalTags": false
}' | jq -r '.data.id' )"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   806    0   163  100   643    714   2819 --:--:-- --:--:-- --:--:--  3535

curl -X GET \
      https://api.eu.opsgenie.com/v2/policies/$ALERT_POLICY_ID?teamId=$TEAM_ID \
     --header "Authorization: GenieKey $OPSGENIE_API_KEY" | jq '.data.responders[].type'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   593    0   593    0     0   4145      0 --:--:-- --:--:-- --:--:--  4176
"escalation"
```

I have submmited a suggestion to edit the API documentation.

**Comment by anjmao** [comment]
Hi, @grubernaut @fnuva @tombuildsstuff @janhorstmann 
Can we merge opened PR for escalation type? We need this field to migrate to terraform.

**Comment by janhorstmann** [comment]
I went ahead and tested this locally and realised that responder type is also validated in [opsgenie-go-sdk-v2](https://github.com/opsgenie/opsgenie-go-sdk-v2). I have also [created an issue](https://github.com/opsgenie/opsgenie-go-sdk-v2/issues/118) and [a simple PR](https://github.com/opsgenie/opsgenie-go-sdk-v2/pull/119) there.
With those changes everything worked as expected.

@frknyldz I have seen that you are doing releases. Could you provide me with some guidance and maybe help on how to move this along?

**Comment by arjunrajpal** [comment]
Hi everyone, this issue has been fixed in [v0.6.27](https://github.com/opsgenie/terraform-provider-opsgenie/releases/tag/v0.6.27). Kindly check !
